components/debounce/core/browser/debounce_rule.cc

```diff
@@ -52,6 +54,34 @@ std::string_view CanonicalizeHostForMatching(std::string_view host_piece) {
   return host_piece;
 }
 
+// Extract the host from |url| using a simple parsing algorithm
+// WARNING: this is a special-purpose function whose output should not be used.
+std::string NaivelyExtractHostnameFromUrl(const std::string& url) {
+  CHECK(GURL(url).SchemeIsHTTPOrHTTPS());
+
+  const std::string kHttp =
+      base::StrCat({url::kHttpScheme, url::kStandardSchemeSeparator});
+  const std::string kHttps =
+      base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator});
+  std::string mutable_url = url;
+
+  if (base::StartsWith(mutable_url, kHttps,
+                       base::CompareCase::INSENSITIVE_ASCII)) {
+    mutable_url = mutable_url.substr(kHttps.length());
+  } else if (base::StartsWith(mutable_url, kHttp,
+                              base::CompareCase::INSENSITIVE_ASCII)) {
+    mutable_url = mutable_url.substr(kHttp.length());
+  }
+
+  const std::vector<std::string> parts = base::SplitString(
+      mutable_url, ":/", base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
```

Comment:
I think this will fail with URLs containing ports OR IPs

Reply:
"Fail" as in "we won't debounce those"?

That seems okay. I don't think we really expect to see those around.

If we cared about that, then I supposed we could split on just `/`?

Reply:
We need to include `:` in the split, due to how `embedded_test_server()->GetURL()` works (see https://source.chromium.org/chromium/chromium/src/+/main:net/test/embedded_test_server/embedded_test_server.h;l=428-430). Unless there is some clever way around that, a lot of our tests will fail.